src/HtmlReader/effects.ts

```diff
@@ -94,7 +94,79 @@ export function setCss(html: HTMLElement, settings: ReaderSettings): void {
     familyToReadiumFamily[settings.fontFamily]
   );
   setCSSProperty(html, '--USER__fontSize', `${settings.fontSize}%`);
+
+  // GLOBAL
   setCSSProperty(html, 'overflow', settings.isScrolling ? 'scroll' : 'hidden');
   // set the number of columns to only ever have 1.
   setCSSProperty(html, '--USER__colCount', '1');
 }
+
+/**
+ * Apply the transform property to the iframe document to fit the current screen viewport.
+ */
+export function setFXLCss(
+  layout: 'fixed' | 'reflowable' | undefined,
+  iframeDocument: Document,
+  iframeContainer: HTMLElement
+): void {
+  if (layout !== 'fixed') return;
+
+  let { contentWidth, contentHeight } = extractContentViewportSize(
+    iframeDocument
+  );
+
+  const { containerWidth, containerHeight } = extractContentContainerSize(
+    iframeContainer
+  );
+
+  // Make it default scale of 1 in case we don't have the content width/height
+  contentWidth = contentWidth ?? containerWidth;
+  contentHeight = contentHeight ?? containerHeight;
+
+  // https://css-tricks.com/scaled-proportional-blocks-with-css-and-javascript/
+  const scale = Math.min(
+    containerWidth / contentWidth,
+    containerHeight / contentHeight
+  );
+
+  setCSSProperty(
+    iframeDocument.documentElement,
+    'transform',
+    `scale(${scale})`
+  );
+  setCSSProperty(
+    iframeDocument.documentElement,
+    'transform-origin',
+    'top center'
+  );
+}
+
+/**
+ * Extract the publication's width and height from the iframe viewport meta tag.
+ */
+function extractContentViewportSize(
+  iframeDocument: Document
+): { contentWidth: number | undefined; contentHeight: number | undefined } {
+  const viewport = iframeDocument?.querySelector('meta[name="viewport"]');
+  const content = viewport?.getAttribute('content');
+
+  const width = content?.match(/width=(\d+)/);
+  const height = content?.match(/height=(\d+)/);
```

Comment:
We trust that the publication will always include this viewport in their HTML files. Though I am unsure if they made this a standard in Epub doc. This only applies to epub2 and I can't find any resources that mention this.

Reply:
Looks like this is what Readium mobile did actually. 
https://github.com/readium/swift-toolkit/blob/b4a9358ad9d0a7cb0072f5c9b1bbb72a7a2d28c7/Sources/Navigator/EPUB/Scripts/src/index-fixed-wrapper.js#L26